core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -184,9 +248,14 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
           if (OperatingSystem.IS_WINDOWS)
             safeForceUnmap()
           raf.setLength(roundedNewSize)
-          _length = roundedNewSize
-          mmap = raf.getChannel().map(FileChannel.MapMode.READ_WRITE, 0, roundedNewSize)
-          _maxEntries = mmap.limit() / entrySize
+          _length = Some(roundedNewSize)
+          _mmap = {
```

Comment:
same here -- more DRY to say:

    _mmap = raf.getChanne.map(if (writable) READ_WRITE else READ_ONLY, 0, roundedNewSize)

Reply:
Fixed.